### DISCUSSION TITLE
Extracting features of several timeseries

### DISCUSSION THREAD

**Discussion by arresejo** [blue-yonder/tsfresh#980] [Q&A] [answered]
Hello,

First of all, congratulations for the amazing work you did !
I'm quite new to the word of time series, sorry in advance if the questions seem too obvious :) 

I'm currently trying to use tsfresh to extract useful features of several currency pairs time series (Forex) to perform some clustering in order to find some correlation between pairs.

So first, I retrieve all the raw data for several currency pairs and append everything to build the dataframe in the format expected by tsfresh: 
![image](https://user-images.githubusercontent.com/3316147/200693332-09cf753e-d41d-4b4d-998c-0bbadb911aef.png)

My questions: 
1)  Is it the good approach to pass directly this big dataframe to _extract_features_ or should I rather use a rolling window approach ? 
2) As I'm working on historical data, is it possible to perform the selection of relevant features ? 
3) Do you recommend to perform some scaling (e.g. min-max) before calling _extract_features_ ?

Thanks in advance, 
Jorge

**Comment by kempa-liehr** [comment, marked as answer]
Hi Jorge,
Thanks for your encouraging feedback. 

1. Passing a time-series of this length to tsfresh would only work for the minimal feature set due to constraints of computational complexity. So, I suggest to use a rolling window.
2. Yes, that is possible, have a look at the following article, which describes the idea of feature selection for clustering: https://ieeexplore.ieee.org/document/9444391
3. Yes, you should consider scaling.

Cheers,
Andreas

**Reply by arresejo** [reply]
Thanks @kempa-liehr, really appreciate ! 

Cheers,
Jorge
